src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -76,103 +110,127 @@
-     * by dependentClass has been seen on the command line.
+     * Called by the command line parser when an argument value contained in the class specified
+     * by {@code predecessorClass} has been seen on the command line.
      *
-     * Return true if the argument is allowed (i.e., this name of this class was specified
-     * as a predecessor on the command line) otherwise false.
+     * Return true if the argument is allowed (i.e., this name of {@code predecessorClass} was
+     * specified as a predecessor on the command line), otherwise false.
      *
      * This method can be used by both the command line parser and the descriptor class for
      * determining when to issue error messages for "dangling" arguments (dependent arguments
      * for which a value has been supplied on the command line, but for which the predecessor
-     * argument was not supplied).
+     * plugin was not supplied).
      *
-     * When this method returns "false", the parser will issue an error message if an argument
-     * value in this class has been set on the command line.
+     * When this method returns "false", the parser will issue an error message.
      *
-     * @param dependentClass
+     * @param predecessorClass class which contains the argument that is the subject of the query
      * @return true if the plugin for this class was specified on the command line, or the
      * values in this class may be set byt he user, otherwise false
      */
-    public abstract boolean isDependentArgumentAllowed(Class<?> dependentClass);
+    public abstract boolean isDependentArgumentAllowed(Class<?> predecessorClass);
 
     /**
      * This method is called after all command line arguments have been processed to allow
-     * the descriptor to validate the plugin arguments that have been specified.
+     * the descriptor to validate and resolve the plugin arguments that have been specified, and to
+     * establish any state necessary to implement {@link #getResolvedInstances}.
      *
-     * It is the descriptor's job to contain an argument list which will be populated
-     * by the command line parser with the name of each plugin specified on the command line,
-     * and for each such plugin, to maintain a list of the corresponding instance. This
-     * method gives the descriptor a chance to reduce that list to include only those
-     * instances actually seen on the command line.
+     * It is the descriptor's job to maintain a list of all plugins discovered by the command
+     * line parser (signaled by calls to {@link #createInstanceForPlugin}), and an @Argument
+     * field to contain name of each plugin specified by the user on the command line. This
+     * method gives the descriptor a chance to validate and resolve those two lists against one
+     * another, and to derive a reduced list that contains only those instances that actually
+     * should be enabled, based on whatever resolution policy is in place for the descriptor.
```

Comment:
Interestingly, the the implementation in GATKReadPluginDescriptor is actually doing none of this merging at this stage but rather at the get instance stage. Indeed this almost seems misleading as for any managed classes with arguments we can't have two instances floating around when we perform command line parsing.

Reply:
Yeah, I think the language is too strong. The first line of this comment says "and to establish any state necessary to implement {@link #getResolvedInstances}", which I think is the essential thing. The language above, where I say "..and resolve those two lists..." is probably to prescriptive, so I'll relax that a bit. It doesn't matter whether the resolution is done here, or in getResolvedInstances; only that once this method is called, getResolvedInstances should have the state necessary to render a resolved list.

This PR is describing a change to the existing contract though, so once we assimilate it into GATK, we'll have to change the implementation of the descriptors to honor the new contract. I think the current GATK read filter descriptor does the "resolution" part of this in "getMergedReadFilter". To conform to this new contract, that will have to be moved into getResolvedInstances. I have a branch where I made those changes to test this all out. Once we've completed the review of this branch, I'll probably run it with that again to make sure it all works.